core/src/main/java/org/infinispan/util/concurrent/BlockingManager.java

```diff
@@ -186,7 +189,8 @@
 
    /**
     * Subscribes to the provided blocking publisher using the the blocking executor, ignoring all elements and returning
-    * a {@link CompletionStage} with a value of null which completes on a non-blocking thread.
+    * a {@link CompletionStage} with a value of null which completes on a non-blocking thread. This method is designed
```

Comment:
Side effect would be something like `doOnNext` etc. Also in regards to subscribe vs request that is a bit pedantic as I was assuming they would be the same, especially since we subscribe using `Integer.MAX_VALUE` For the request. Did you want me to write more info for that?

Reply:
I wouldn't add more info, I'd rather remove the "This method is designed ..." sentence completely because I don't think it adds much. If `doOnNext()` qualifies as "when subscribed", then I assume anything the publisher does qualifies.

I would also change the next sentences a bit. Assuming we do need the method to behave differently depending on the caller thread (similar to the `blockingPublisher()` discussion), we should be upfront about it and say "If the current thread is X, then Y happens." instead of saying Y "always" happens and then backtracking on that promise.

In general, I would also like it if the javadoc was using the reactive streams terms more precisely, e.g. instead of saying "everything including subscription, publication and collection of values will be done on the current thread" I'd say "the publisher is subscribed on the current thread and the values are not observed on any particular thread".